README.md

```diff
@@ -98,142 +113,140 @@ oc login -u developer -p developer https://localhost:8443
 
 In the OpenShift console, you can see the different deployments for the various EnMasse components. You can go into each pod and look at the logs. If we go to the address controller log, you can see that its creating a 'default' address space.
 
-#### Authentication and Authorization
+## (Optional) Installing Apache Spark
 
-Go to the OpenShift console, application -> routes, and click on the hostname for the 'keycloak' route. This should bring you to the keycloak admin console. The admin user is protected by the password that was set in the playbook.
+An official support for Apache Spark on OpenShift is provided by the [radanalytics.io](https://radanalytics.io/) project by means of
+the Oshinko application with a Web UI for deploying a Spark cluster. Other than using such a Web UI, a CLI tool is available as well which is used for this workshop.
 
-In the Keycloak UI, make sure you are in the 'default' realm. Then create a new user, and a set of credentials for that user. Make sure the user is
-enabled, and that the credentials are not marked as temporary.
+Go to [Oshinko CLI downloads](https://github.com/radanalyticsio/oshinko-cli/releases) and download
+the latest release (0.3.1 as of time of writing). Unpack the release:
 
-For this workshop we could use following users for example :
+```
+tar xvf oshinko_v0.3.1_linux_amd64.tar.gz
+```
 
-* _console_ : as a "real" user who uses the console for creating addresses
-* _deviceX_ : as deviceX (i.e. device1, device2, ...) user who can only send/recv from a particular address
-* _sparkdriver_ : as Spark driver application user 
-* _thermostat_ : as thermostat application user
+Then use the following command in order to deploy a Spark cluster made by one master node and one slave node.
 
-It's clear that this workshop involves a "real" user who is in charge to create and configure the addresses and some other users which are related to applications running in the cluster and devices running in the field.
+```
+export SPARK_NAME=<something>
+
+oc new-project spark
+./oshinko_linux_amd64/oshinko create $SPARK_NAME --masters=1 --workers=1
+```
+
+## IoT Application
+
+Now that the cluster has been set up, its time to work on the example application. First, we will
+provision messaging infrastructure to use with the application.
+
+### Provisioning messaging
+
+Go to the OpenShift Console.
 
-It is now time to create some authorization rules to ensure the different users can only access the
-addresses assigned. This is done by creating groups with names on the form `send_myqueue` and have a
-user join that group to gain the permission (to send messages to address `myqueue`). We can for
-example create the following groups that will match the addresses that we will create later:
+In the OpenShift Service Catalog overview, select either of "EnMasse (standard)" or "EnMasse (brokered)".
```

Comment:
should we drive the attendees to select one of the them and not let they chose what to use ?
Then just a question from me : is it possible to create an "anycast" address if the address space is the brokered one ? I know that in any case we have the routers network but I thought that it was disabled in some way (we have the "max" address which is anycast).

Reply:
Yeah, that is a good point. We should steer them to 'standard' probably. I was just thinking that you could potentially replace 'anycast' with 'queue' for the brokered address space.

Reply:
I would go for the "standard" without changing anything and showing another kind of address : a direct messaging anycast address which is something "new". But it's your call ... you are the lead ! ;)